File: backend/src/main/java/com/eskisehir/eventapi/dto/UserResponse.java
package com.eskisehir.eventapi.dto;

import java.time.LocalDateTime;

/**
 * Response DTO for user information.
 */
public class UserResponse {

    private Long id;
    private String email;
    private String displayName;
    private LocalDateTime createdAt;
    private LocalDateTime lastLoginAt;

    public UserResponse() {}

    public UserResponse(Long id, String email, String displayName, LocalDateTime createdAt, LocalDateTime lastLoginAt) {
        this.id = id;
        this.email = email;
        this.displayName = displayName;
        this.createdAt = createdAt;
        this.lastLoginAt = lastLoginAt;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getLastLoginAt() {
        return lastLoginAt;
    }

    public void setLastLoginAt(LocalDateTime lastLoginAt) {
        this.lastLoginAt = lastLoginAt;
    }
}

File: mobile/app/src/main/java/com/eskisehir/eventapp/data/model/UserResponse.kt
package com.eskisehir.eventapp.data.model

import com.google.gson.annotations.SerializedName

/**
 * Response DTO for user information.
 */
data class UserResponse(
    @SerializedName("id")
    val id: Long,
    @SerializedName("email")
    val email: String,
    @SerializedName("displayName")
    val displayName: String,
    @SerializedName("createdAt")
    val createdAt: String,
    @SerializedName("lastLoginAt")
    val lastLoginAt: String? = null
)
